src/fs.rs

```diff
@@ -83,16 +84,37 @@ impl Responder for NamedFile {
     type Item = HttpResponse;
     type Error = io::Error;
 
-    fn respond_to(mut self, _: HttpRequest) -> Result<HttpResponse, io::Error> {
-        let mut resp = HTTPOk.build();
-        resp.content_encoding(ContentEncoding::Identity);
-        if let Some(ext) = self.path().extension() {
+    fn respond_to(mut self, req: HttpRequest) -> Result<HttpResponse, io::Error> {
+        if let Some(rangeheader) = req.headers().get(header::RANGE) {
+            let file_metadata = metadata(self.0)?;
+            if let Ok(ranges) = HttpRange::parse(rangeheader.to_str().unwrap(), file_metadata.len()) {
```

Comment:
you should not "unwrap" here, in case of error you should return `HTTPRangeNotSatisfiable`